test/tests_for_sdkmr/ee2_SDKMethodRunner_test_EE2Runjob_test.py

```diff
@@ -9,17 +9,17 @@
 import requests_mock
 from mock import MagicMock
 
-from lib.execution_engine2.db.MongoUtil import MongoUtil
-from lib.execution_engine2.db.models.models import Job
-from lib.execution_engine2.sdk.SDKMethodRunner import SDKMethodRunner
-from lib.execution_engine2.utils.CondorTuples import SubmissionInfo
+from execution_engine2.exceptions import CannotRetryJob, RetryFailureException
+from execution_engine2.sdk.job_submission_parameters import JobRequirements
```

Comment:
Unless I'm missing something it looks like quite a bit of the unhappy path code isn't tested - like updating the batch and retry parent jobs and some of the validation code.

Reply:
Added some more tests. They are now tested as per https://codecov.io/github/kbase/execution_engine2/commit/38d917f5db5293a1bd93b7c4fb25619e0698fc23

Reply:
Maybe I'm missing something but that seems to show that quite a bit of the unhappy paths in the retry code are untested.

Reply:
@MrCreosote Hmm that's strange.. I thought it did.  What about this one https://app.codecov.io/gh/kbase/execution_engine2/compare/383/tree/lib/execution_engine2/sdk/EE2Runjob.py

Reply:
That's really weird, I'm clearly testing `Cannot retry batch job parents. Must retry individual jobs`  in `test_validate_retry` but it shows it as untested.

Reply:
Ok the second link looks a lot better

Reply:
I feel like this case is important enough to warrant a test though: https://app.codecov.io/gh/kbase/execution_engine2/compare/383/tree/lib/execution_engine2/sdk/EE2Runjob.py#D1L659

Reply:
> That's really weird, I'm clearly testing Cannot retry batch job parents. Must retry individual jobs in test_validate_retry but it shows it as untested.

For me it shows as tested in the 2nd link

Reply:
This case is so simple to test you might as well: https://app.codecov.io/gh/kbase/execution_engine2/compare/383/tree/lib/execution_engine2/sdk/EE2Runjob.py#D1L631

Reply:
Ugh, codecov doesn't seem to scroll to the line numbers in the links

Reply:
done

Reply:
:+1: Looks like codecov hasn't caught up with today's commits yet though